packager/react-packager/src/DependencyResolver/DependencyGraph/ResolutionRequest.js

```diff
@@ -336,6 +336,8 @@ class ResolutionRequest {
         file = potentialModulePath + '.native.js';
```

Comment:
Another late late chime in: An argument _for_ supporting `jsx` is that up until recently (e.g. pre-flow), `.js` pretty consistently meant "a version of ES that either matches current or future ES specs".  I'm assuming that JSX will never become part of ES proper?

Similarly, why not `.flow`? There is `.ts` after all. (aside from it being potentially too late to bring up)

File extensions help tools and infrastructure determine what to do w/ source. Having to guess is error prone and added complication